app_dart/lib/src/request_handlers/push_gold_status_to_github.dart

```diff
@@ -0,0 +1,274 @@
+import 'package:gcloud/db.dart';
+import 'package:github/server.dart';
+import 'package:graphql/client.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../model/appengine/github_gold_status_update.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../service/datastore.dart';
+import 'refresh_cirrus_status.dart';
+
+@immutable
+class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
+  PushGoldStatusToGithub(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting DatastoreServiceProvider datastoreProvider,
+    @visibleForTesting LoggingProvider loggingProvider,
+    HttpClient goldClient,
+  })  : datastoreProvider =
+            datastoreProvider ?? DatastoreService.defaultProvider,
+        loggingProvider = loggingProvider ?? Providers.serviceScopeLogger,
+        goldClient = goldClient ?? HttpClient(),
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+  final LoggingProvider loggingProvider;
+  final HttpClient goldClient;
+
+  static const String kTargetUrl = 'https://flutter-gold.skia.org/changelists';
+  static const String kContext = 'flutter-gold';
+
+  @override
+  Future<Body> get() async {
+    final Logging log = loggingProvider();
+    final DatastoreService datastore = datastoreProvider();
+
+    if (authContext.clientContext.isDevelopmentEnvironment) {
+      // Don't push gold status from the local dev server.
+      return Body.empty;
+    }
+
+    const RepositorySlug slug = RepositorySlug('flutter', 'flutter');
+    final GitHub gitHubClient = await config.createGitHubClient();
+    final GraphQLClient cirrusClient = await config.createCirrusGraphQLClient();
+    final List<GithubGoldStatusUpdate> statusUpdates =
+        <GithubGoldStatusUpdate>[];
+    final List<String> cirrusCheckStatuses = <String>[];
+    const List<String> cirrusInProgressStates = <String>[
+      'EXECUTING',
+      'CREATED',
+      'TRIGGERED',
+      'NEEDS_APPROVAL'
+    ];
+
+    await for (PullRequest pr in gitHubClient.pullRequests.list(slug)) {
+      // Check run statuses for this pr
+      final List<dynamic> cirrusStatuses =
+          await queryCirrusGraphQL(pr.head.sha, cirrusClient, log, 'flutter');
+      for (dynamic runStatus in cirrusStatuses) {
+        final String status = runStatus['status'];
+        final String taskName = runStatus['name'];
+
+        log.debug(
+            'Found Cirrus build status for pull request ${pr.number}, commit '
+            '${pr.head.sha}: $taskName ($status)');
+
+        if (taskName.contains('framework')) {
+          // Analyzer requires curly braces here.
```

Comment:
nit: I'm not sure we need the comment here.

Also, to avoid getting conflicts between analysis rules here and dartfmt, we could consider adding `curly_braces_in_flow_control_structures` to the analysis_options.yaml, which would then enforce that you use curly braces and always make dartfmt do what we want here.  Just don't tell Hixie ;)

Reply:
https://github.com/flutter/cocoon/pull/661 :)

Reply:
That update has landed.